ArcadiaParties.API/Startup.cs

```diff
@@ -36,11 +39,16 @@ public void ConfigureServices(IServiceCollection services)
             services.AddSwaggerGen(c =>
             {
                 c.SwaggerDoc("v1", new OpenApiInfo { Title = "ArcadiaParties.API", Version = "v1" });
+                c.EnableAnnotations();
             });
 
             services.AddMediatR(typeof(SeedCommand).Assembly);
 
             services.AddAuthentication(NegotiateDefaults.AuthenticationScheme).AddNegotiate();
+
+            services.AddAutoMapper(typeof(UserRepository).Assembly);
+
+            services.AddScoped<IUserRepository, UserRepository>();
```

Comment:
As you implemented repository pattern, you can decide to move logic of getting roles from middleware to another repository. Or, to be more accurate, to own query.

Reply:
Ok, you already use query in middleware, but directly use context in query. I already explained cons of such approach.

Reply:
Now we have chain: middleware -> query -> repository -> context